### PR TITLE
Make Wake Word detection modular

### PR DESCRIPTION
**Pull request by MatthewScholefield**
This separates Pocketsphinx into its own class allowing other wake word recognizes to be switched in easier.

### CODE REVIEW

**Comment by hersche** [comment]
could this step may closing #842 since the second module could use snowboy or so?
i dislike to reinvent the wheel more than necessary ;)

**Comment by MatthewScholefield** [comment]
@forslund Fixed! :)

File: mycroft/client/speech/recognizer/pocketsphinx_recognizer.py [hunk 1]
```diff
@@ -19,16 +19,18 @@
 import os
 import tempfile
 import time
-from os.path import join, dirname, abspath
 
+from os.path import join, dirname, abspath
 from pocketsphinx import Decoder
 
+from mycroft.client.speech.local_recognizer import LocalRecognizer
```
Comment (forslund) [inline mycroft/client/speech/recognizer/pocketsphinx_recognizer.py]: You missed a tiny thing =)

This should be
```python
from mycroft.client.speech.recognizer.local_recognizer import LocalRecognizer
```

Comment (MatthewScholefield) [inline mycroft/client/speech/recognizer/pocketsphinx_recognizer.py]: Good catch! Fixed.

**Comment by forslund** [comment]
Looking good!

**Pull request merged**